### PR TITLE
Add public doc for scheduler

### PR DESCRIPTION
**Pull request by noCharger**
### Description
Add public doc for scheduler

### Issues Resolved

https://github.com/opensearch-project/documentation-website/issues/8263

### Version
2.17

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by noCharger** [comment]
> Thank you for submitting your PR. The PR states are In progress (or Draft) -> Tech review -> Doc review -> Editorial review -> Merged.
> 
> Before you submit your PR for doc review, make sure the content is technically accurate. If you need help finding a tech reviewer, tag a [maintainer](https://github.com/opensearch-project/documentation-website/blob/main/MAINTAINERS.md).
> 
> **When you're ready for doc review, tag the assignee of this PR**. The doc reviewer may push edits to the PR directly or leave comments and editorial suggestions for you to address (let us know in a comment if you have a preference). The doc reviewer will arrange for an editorial review.

@kolchfa-aws

File: _dashboards/management/scheduled-query-acceleration [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+With SQA, you have more flexible scheduling options. You can set refresh intervals based on your specific requirements, ensuring efficient data management without overburdening driver nodes. This means greater control over resource usage and refresh frequency that better aligns with your needs.
+
+### Simplified Index Management
+
+SQA also simplifies index management by allowing you to adjust index settings without multiple, complicated queries. You can easily update refresh intervals or other parameters, thereby streamlining your workflow and reducing manual effort.
+
+## Getting Started
+
+### Prerequisites
+
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration:40]: Also the document of prereq:

Step1:  for setup data source https://opensearch.org/docs/latest/dashboards/management/S3-data-source/, after then dashboard user can fire query via workbench, ref https://opensearch.org/docs/latest/dashboards/management/query-data-source/

Step2: user can accelerate query using secondary index ref https://opensearch.org/docs/latest/dashboards/management/accelerate-external-data/

During this process, user can change config to try out everything described on this doc

File: _dashboards/management/scheduled-query-acceleration [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+#### Configure Async Query External Scheduler Interval
+
+This setting specifies the interval at which the external scheduler checks for tasks, allowing users to customize how often the scheduler initiates refresh operations. Adjusting this interval based on workload requirements can optimize resource usage and control the frequency of driver node operations, helping to manage costs effectively.
+
+[Setting Link](https://github.com/opensearch-project/sql/blob/main/docs/user/admin/settings.rst#pluginsqueryexecutionengineasync_queryexternal_schedulerinterval)
+
+### Spark Configurations
+
+- `spark.flint.job.externalScheduler.enabled`: Default is `false`. Enable external scheduler for Flint auto-refresh to schedule refresh jobs outside of Spark.
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration:71]: on sql plugin 2.17 this value is passsed as true by default

Comment (kolchfa-aws) [reply _dashboards/management/scheduled-query-acceleration]: So is the default for `spark.flint.job.externalScheduler.enabled` true then?

**Comment by kolchfa-aws** [comment]
@noCharger Please review the changes and let me know if anything should be changed. Thank you!

File: _dashboards/management/scheduled-query-acceleration.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+parent: Data sources
+nav_order: 18
+has_children: false
+---
+
+# Scheduled Query Acceleration
+Introduced 2.17
+{: .label .label-purple }
+
+Scheduled Query Acceleration (SQA) is designed to optimize direct queries from OpenSearch to Amazon Simple Storage Service (Amazon S3). It addresses issues often faced when managing and refreshing indexes, views, and data in an automated way.
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:13]: > OpenSearch to Amazon Simple Storage Service (Amazon S3)

This direct query is not only limited to s3, let's use "external data sources" on this page for consistency with other pages

File: _dashboards/management/scheduled-query-acceleration.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+Scheduled Query Acceleration (SQA) is designed to optimize direct queries from OpenSearch to Amazon Simple Storage Service (Amazon S3). It addresses issues often faced when managing and refreshing indexes, views, and data in an automated way. 
+
+Query acceleration is facilitated by secondary indexes like [skipping indexes]({{site.url}}{{site.baseurl}}/dashboards/management/accelerate-external-data/#skipping-indexes), [covering indexes]({{site.url}}{{site.baseurl}}/dashboards/management/accelerate-external-data/#covering-indexes), or [materialized views]({{site.url}}{{site.baseurl}}/dashboards/management/accelerate-external-data/#materialized-views). When queries run, they use these indexes instead of directly querying S3. 
+
+The secondary indexes need to be refreshed periodically to stay current with the Amazon S3 data. This refresh can be scheduled using an internal scheduler (within Spark) or an external scheduler.
+
+Using SQA provides the following benefits:
+
+- **Cost reduction through optimized resource usage**: SQA reduces the operational load on driver nodes, lowering the costs associated with maintaining auto-refresh for indexes and views.
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:21]: there's an experiment with data can be visualized in charts, shall we add them somewhere?

File: _dashboards/management/scheduled-query-acceleration.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+- **Improved observability of refresh operations**: SQA provides visibility into index states and refresh timings, offering insights into data processing and the current system state.
+
+- **Better control over refresh scheduling**: SQA allows flexible scheduling of refresh intervals, helping manage resource usage and refresh frequency according to specific requirements.
+
+- **Simplified index management**: SQA enables updates to index settings, such as refresh intervals, without requiring multiple queries, simplifying workflows.
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:27]: This one may need some clarification, for example originally it was two passes
1. stop the streaming query
2. change the interval and other index options

now it's within a single query

File: _dashboards/management/scheduled-query-acceleration.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+
+## Concepts
+
+Before configuring SQA, familiarize yourself with the following topics:
+
+- [Optimizing query performance using OpenSearch indexing]({{site.url}}{{site.baseurl}}/dashboards/management/accelerate-external-data/)
+- [Flint index refresh](https://github.com/opensearch-project/opensearch-spark/blob/main/docs/index.md#flint-index-refresh)
+- [Index State Management]({{site.url}}{{site.baseurl}}/im-plugin/ism/index/)
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:35]: this is unrelated, the index state is here https://github.com/opensearch-project/opensearch-spark/blob/main/docs/index.md#index-state-transition-1

File: _dashboards/management/scheduled-query-acceleration.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+- Ensure you're running OpenSearch version 2.17 or later.
+- Ensure you have the SQL plugin installed. The SQL plugin is part of most OpenSearch distributions. For more information, see [Installing plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/).
+- Ensure you have configured an Amazon S3 and Amazon EMR Serverless (needed for access to Apache Spark). 
+
+## Configuring SQA
+
+To configure SQA, perform the following steps.
+
+### Step 1: Configure the OpenSearch cluster settings
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:49]: FYI this is enabled by default

File: _dashboards/management/scheduled-query-acceleration.md [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+      "transient": {
+        "plugins.query.executionengine.async_query.external_scheduler.interval": "10 minutes"
+      }
+    }
+    ```
+    {% include copy-curl.html %}
+
+    For more information, see [Settings](https://github.com/opensearch-project/sql/blob/main/docs/user/admin/settings.rst#pluginsqueryexecutionengineasync_queryexternal_schedulerinterval).
+
+### Step 2: Configure Apache Spark settings
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:79]: this step doesn't need

File: _dashboards/management/scheduled-query-acceleration.md [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+
+You can run accelerated queries in [Query Workbench]({{site.url}}{{site.baseurl}}/dashboards/query-workbench/). To run an accelerated query, use the following syntax:
+
+```sql
+CREATE SKIPPING INDEX example_index
+WITH (
+    auto_refresh = true,
+    refresh_interval = '15 minutes',
+    scheduler_mode = 'external'
+);
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:111]: by default it would use external scheduler, no need to mention unless user wants to use internal one

File: _dashboards/management/scheduled-query-acceleration.md [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+## Modifying refresh settings
+
+To modify refresh settings, use the `ALTER` command:
+
+```sql
+ALTER INDEX example_index
+WITH (refresh_interval = '30 minutes');
+```
+{% include copy.html %}
+
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:160]: there are some overlaps between ` ## Managing scheduled jobs` and `## Managing scheduled jobs`

File: _dashboards/management/scheduled-query-acceleration.md (hunk 4 above)
Comment (kolchfa-aws) [reply _dashboards/management/scheduled-query-acceleration.md:21]: Sure, please make a suggestion or commit the updates directly. Thanks!

File: _dashboards/management/scheduled-query-acceleration.md (hunk 5 above)
Comment (kolchfa-aws) [reply _dashboards/management/scheduled-query-acceleration.md:27]: Updated.

File: _dashboards/management/scheduled-query-acceleration.md (hunk 7 above)
Comment (kolchfa-aws) [reply _dashboards/management/scheduled-query-acceleration.md:49]: I put "Set `plugins.query.executionengine.async_query.enabled` to `true` (default value)" so users know it's default. But for `plugins.query.executionengine.async_query.external_scheduler.interval` there's no default value - correct? So users must set it manually.

Comment (noCharger) [reply _dashboards/management/scheduled-query-acceleration.md:49]: > But for plugins.query.executionengine.async_query.external_scheduler.interval there's no default value - correct? So users must set it manually.

Correct, if this value is empty, from spark side (opensearch-spark) it would have an default value (5 mins)

File: _dashboards/management/scheduled-query-acceleration.md [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+    ```
+    {% include copy-curl.html %}
+
+    For more information, see [Settings](https://github.com/opensearch-project/sql/blob/main/docs/user/admin/settings.rst#pluginsqueryexecutionengineasync_queryexternal_schedulerinterval).
+
+### Step 2: Configure a data source
+
+Connect OpenSearch to your Amazon S3 data source using the OpenSearch Dashboards interface. For more information, see [Connecting Amazon S3 to OpenSearch]({{site.url}}{{site.baseurl}}/dashboards/management/S3-data-source/).
+
+After this step, you can directly query your S3 data (the primary data source) using [Query Workbench]({{site.url}}{{site.baseurl}}/dashboards/query-workbench/).
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:83]: Wondering if it's more suitable to say data source is an prereq

File: _dashboards/management/scheduled-query-acceleration.md [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+
+### Step 3: Configure query acceleration
+
+Configure a skipping index, covering index, or materialized view. These secondary data sources are additional data structures that improve query performance by optimizing queries on external data sources, such as Amazon S3. For more information, see [Optimize query performance using OpenSearch indexing]({{site.url}}{{site.baseurl}}/dashboards/management/accelerate-external-data/).
+
+After this step, you can [run accelerated queries](#running-an-accelerated-query) using one of the secondary data sources.
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:89]: This step 3 is the same as "run accelerated queries".  Open source user should be able to use this feature without any customization since default is enabled with threshold as 5 mins. But they are able to custermize via  `Configure the OpenSearch cluster settings`

File: _dashboards/management/scheduled-query-acceleration.md [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+
+```sql
+CREATE SKIPPING INDEX example_index
+WITH (
+    auto_refresh = true,
+    refresh_interval = '15 minutes',
+    scheduler_mode = 'external'
+);
+```
+{% include copy.html %}
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:150]: this seems also duplicate to `## Running an accelerated query
`

File: _dashboards/management/scheduled-query-acceleration.md [hunk 14]
```diff
@@ -0,0 +1,252 @@
...
+
+## Managing scheduled jobs
+
+Use the following commands to manage scheduled jobs.
+
+### Enabling jobs
+
+To disable the external scheduler, use the ALTER command with a manual refresh:
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:167]:
```suggestion
To disable the auto refresh with internal / external scheduler, use the ALTER command with a manual refresh:
```

File: _dashboards/management/scheduled-query-acceleration.md [hunk 15]
```diff
@@ -0,0 +1,252 @@
...
+
+```sql
+ALTER MATERIALIZED VIEW myglue_test.default.count_by_status_v9 WITH (auto_refresh = false);
+```
+{% include copy.html %}
+
+To enable the external scheduler, use the ALTER command with an auto-refresh:
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:174]:
```suggestion
To enable the auto refresh with internal / external scheduler, use the ALTER command with an auto-refresh:
```

File: _dashboards/management/scheduled-query-acceleration.md [hunk 16]
```diff
@@ -0,0 +1,252 @@
...
+
+```sql
+ALTER INDEX example_index
+WITH (refresh_interval = '30 minutes');
+```
+{% include copy.html %}
+
+### Updating the scheduler mode
+
+To update the scheduler mode, specify the `scheduler_mode` in the `WITH` clause:
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:193]:
```suggestion
To switch the scheduler mode from internal to external, or vice versa, specify the `scheduler_mode` in the `WITH` clause:
```

File: _dashboards/management/scheduled-query-acceleration.md [hunk 17]
```diff
@@ -0,0 +1,252 @@
...
+{% include copy.html %}
+
+### Verifying scheduler job status
+
+To verify scheduler job status, use the following request:
+
+```json
+GET /.async-query-scheduler/_search
+```
+{% include copy-curl.html %}
```
Comment (noCharger) [inline _dashboards/management/scheduled-query-acceleration.md:207]:
```suggestion
### Inspect scheduler metadata

To inspect scheduler metadata, use the following dsl request:

```json
GET /.async-query-scheduler/_search
```
{% include copy-curl.html %}
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Given that you received additional comments from @noCharger prior to editorial, tag me on any resulting additions/modifications, and I can give them a quick read. Thanks!

File: _dashboards/management/scheduled-query-acceleration.md [hunk 18]
```diff
@@ -0,0 +1,252 @@
...
+
+You can validate your settings by running a test query and verifying the scheduler configurations:
+
+```sql
+SHOW FLINT INDEXES EXTENDED
+```
+{% include copy.html %}
+
+For more information, see [OpenSearch Spark documentation](https://github.com/opensearch-project/opensearch-spark/blob/main/docs/index.md#all-indexes).
+
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/management/scheduled-query-acceleration.md:245]: Yes.

File: _dashboards/management/scheduled-query-acceleration.md [hunk 19]
```diff
@@ -39,16 +39,13 @@ Before configuring SQA, familiarize yourself with the following topics:
 Before configuring SQA, verify that the following requirements are met:
 
 - Ensure you're running OpenSearch version 2.17 or later.
-- Ensure you have the SQL plugin installed. The SQL plugin is part of most OpenSearch distributions. For more information, see [Installing plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/).
-- Ensure you have configured an Amazon S3 and Amazon EMR Serverless (needed for access to Apache Spark). 
+- Ensure you have the SQL plugin installed. The SQL plugin is included in most OpenSearch distributions. For more information, see [Installing plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/).
+- Ensure you have configured a data source (in this example, Amazon S3): Configure a skipping index, covering index, or materialized view. These secondary data sources are additional data structures that improve query performance by optimizing queries sent to  external data sources, such as Amazon S3. For more information, see [Optimizing query performance using OpenSearch indexing]({{site.url}}{{site.baseurl}}/dashboards/management/accelerate-external-data/).
```
Comment (natebower) [inline _dashboards/management/scheduled-query-acceleration.md:43]: Extra space between "to external"

File: _dashboards/management/scheduled-query-acceleration.md [hunk 20]
```diff
@@ -164,42 +135,35 @@ Use the following commands to manage scheduled jobs.
 
 ### Enabling jobs
 
-To disable the external scheduler, use the ALTER command with a manual refresh:
+To disable auto refresh using an internal or external scheduler, set `auto_refresh` to `false`:
```
Comment (natebower) [inline _dashboards/management/scheduled-query-acceleration.md:138]: I believe there may be other instances where "auto-refresh" is hyphenated.

File: _dashboards/management/scheduled-query-acceleration.md [hunk 21]
```diff
@@ -212,41 +176,33 @@ We recommend the following best practices when using SQA.
 
 ### Performance optimization
 
-We recommend the following practices for best performance:
-
-- **Recommended refresh intervals**: Choosing the right refresh interval is crucial for balancing resource usage and system performance. Consider your workload requirements and the freshness of data you need when setting intervals.
+- **Recommended refresh intervals**: Choosing the right refresh interval is crucial for balancing resource usage and system performance. Consider your workload requirements and the freshness of the data you need when setting intervals.
 
-- **Concurrent job limits**: Limit the number of concurrent jobs running to avoid overloading system resources. Monitor system capacity and adjust job limits accordingly to ensure optimal performance.
+- **Concurrent job limits**: Limit the number of concurrent running jobs running to avoid overloading system resources. Monitor system capacity and adjust job limits accordingly to ensure optimal performance.
```
Comment (natebower) [inline _dashboards/management/scheduled-query-acceleration.md:181]: Delete the instance of "running" after "jobs".

File: _dashboards/management/scheduled-query-acceleration.md [hunk 22]
```diff
@@ -0,0 +1,252 @@
...
+
+- **Concurrent job limits**: Limit the number of concurrent jobs running to avoid overloading system resources. Monitor system capacity and adjust job limits accordingly to ensure optimal performance.
+
+- **Resource usage**: Efficient resource allocation is key to maximizing performance. Properly allocate memory, CPU, and I/O based on the workload and the type of queries you're running.
+
+### Cost management
+
+We recommend the following practices to reduce costs:
+
+- **Use an external scheduler**: An external scheduler offloads refresh operations, reducing the demand on core driver nodes.
```
[the remark this answers is not in the document]
Comment (natebower) [reply _dashboards/management/scheduled-query-acceleration.md]: @kolchfa-aws Looks like this one may not have made it.

[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/management/scheduled-query-acceleration.md]: It's in a separate commit https://github.com/opensearch-project/documentation-website/pull/8825/commits/3783d96ac4cd34eebff187a674d6c3824159c8b9

**Review by natebower — APPROVED** [review]
LGTM
